diff --git a/src/display.c b/src/display.c
index e09aa34..e856971 100644
--- a/src/display.c
+++ b/src/display.c
@@ -538,7 +538,7 @@ start_user_session (Display *display, const gchar *session, const gchar *languag
             if (display->priv->session_wrapper)
             {
                 gchar *old_command = session_command;
-                session_command = g_strdup_printf ("%s '%s'", display->priv->session_wrapper, session_command);
+                session_command = g_strdup_printf ("%s '%s'", display->priv->session_wrapper, session);
                 g_free (old_command);
             }
             display->priv->user_session = session_new ();
